src/Bicep.Core/Workspaces/ProviderDescriptorBundle.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem.Providers;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+
+namespace Bicep.Core.Workspaces;
+
+public record ProviderDescriptorBundle(
+    IEnumerable<ResultWithDiagnostic<ResourceTypesProviderDescriptor>> ImplicitProviders,
```

Comment:
Why a mix of enumerable + immutable here?

Reply:
I wasn't sure what I was doing, figured it out now